src/Algorithm.ts

```diff
@@ -9,22 +11,61 @@ export default class Algorithm extends Builder {
     context.inAlg = true;
     const { node } = context;
 
-    if ('ecmarkdownOut' in node) {
-      // i.e., we already parsed this during the lint phase
-      // @ts-ignore
-      node.innerHTML = node.ecmarkdownOut.replace(/((?:\s+|>)[!?])\s+(\w+\s*\()/g, '$1&nbsp;$2');
-      return;
-    }
+    // prettier-ignore
+    const rawHtml =
+      'ecmarkdownOut' in node
+        ? (node as any).ecmarkdownOut
+        : emd.algorithm(node.innerHTML);
 
     // replace spaces after !/? with &nbsp; to prevent bad line breaking
-    const html = emd
-      .algorithm(node.innerHTML)
-      .replace(/((?:\s+|>)[!?])\s+(\w+\s*\()/g, '$1&nbsp;$2');
+    const html = rawHtml.replace(/((?:\s+|>)[!?])\s+(\w+\s*\()/g, '$1&nbsp;$2');
     node.innerHTML = html;
+
+    let labeledStepEntries: StepBiblioEntry[] = [];
+    let replaces = node.getAttribute('replaces-step');
```

Comment:
Should we remove this attribute from the output?

Reply:
Ecmarkup doesn't really clean up after itself, as a rule - all the `emu-whatever` tags are in the output, etc.

Reply:
Fine. We should reconsider that.

Reply:
Opened https://github.com/tc39/ecmarkup/issues/218 to track.